include/aws/event-stream/event_stream.h

```diff
@@ -65,9 +65,13 @@ struct aws_event_stream_message {
     struct aws_byte_buf message_buffer;
 };
 
-#define AWS_EVENT_STREAM_PRELUDE_LENGTH (uint32_t)(sizeof(uint32_t) + sizeof(uint32_t) + sizeof(uint32_t))
-
-#define AWS_EVENT_STREAM_TRAILER_LENGTH (uint32_t)(sizeof(uint32_t))
+#if SIZE_MAX == UINT32_MAX
+#    define AWS_EVENT_STREAM_PRELUDE_LENGTH (sizeof(uint32_t) + sizeof(uint32_t) + sizeof(uint32_t))
+#    define AWS_EVENT_STREAM_TRAILER_LENGTH (sizeof(uint32_t))
+#else /* cast to uint32_t */
+#    define AWS_EVENT_STREAM_PRELUDE_LENGTH (uint32_t)(sizeof(uint32_t) + sizeof(uint32_t) + sizeof(uint32_t))
+#    define AWS_EVENT_STREAM_TRAILER_LENGTH (uint32_t)(sizeof(uint32_t))
+#endif
```

Comment:
I don't understand, why not make these a simple constant like 
#define AWS_EVENT_STREAM_PRELUDE_LENGTH 12

Reply:
👍 yeah, that would be simpler, at this point